uc-crux-llvm/src/UCCrux/LLVM/Overrides.hs

```diff
@@ -0,0 +1,184 @@
+registerUnsoundOverrides proxy llvmModule mtrans =
+  do
+    let llvmCtx = mtrans ^. transContext
+    let ?lc = llvmCtx ^. llvmTypeCtx
+    register_llvm_overrides llvmModule [] (unsoundOverrides proxy) llvmCtx
+
+------------------------------------------------------------------------
+
+-- ** Declarations
+
+unsoundOverrides ::
+  ( IsSymInterface sym,
+    HasLLVMAnn sym,
+    ArchOk arch,
+    ?lc :: TypeContext,
+    HasModel personality
+  ) =>
+  proxy arch ->
+  [OverrideTemplate (personality sym) sym arch rtp l a]
+unsoundOverrides arch =
+  -- NOTE(lb): The list here should match that in the "Config" module.
+  [ basic_llvm_override $
+      [llvmOvr| i32 @gethostname( i8* , size_t ) |]
+        (\memOps sym args -> Ctx.uncurryAssignment (callGetHostName arch sym memOps) args),
+    basic_llvm_override $
+      [llvmOvr| i8* @getenv( i8* ) |]
+        (\memOps sym args -> Ctx.uncurryAssignment (callGetEnv arch sym memOps) args)
+  ]
+
+------------------------------------------------------------------------
+
+-- ** Implementations
+
+callGetHostName ::
+  ( IsSymInterface sym,
+    HasLLVMAnn sym,
+    wptr ~ ArchWidth arch,
+    ArchOk arch,
+    ?lc :: TypeContext
+  ) =>
+  proxy arch ->
+  sym ->
+  GlobalVar Mem ->
+  RegEntry sym (LLVMPointerType wptr) ->
+  RegEntry sym (BVType wptr) ->
+  OverrideSim p sym ext r args ret (RegValue sym (BVType 32))
+callGetHostName _proxy sym mvar (regValue -> ptr) (regValue -> len) =
+  do
+    let hostname = "hostname"
+    let lenLt bv = liftIO (What4.bvSlt sym len =<< What4.bvLit sym ?ptrWidth bv)
+    lenNeg <- lenLt (BV.mkBV ?ptrWidth 0)
+    -- TODO(lb): Panic if ?ptrWidth is like 2 or something crazy
+    lenSmall <- lenLt (BV.mkBV ?ptrWidth (fromIntegral (BS.length hostname)))
+    Override.symbolicBranches
+      emptyRegMap
+      [ ( lenNeg,
+          Override.modifyGlobal mvar $ \_mem ->
+            unimplemented "callGetHostName" Unimplemented.GetHostNameNegativeSize,
+          Nothing
+        ),
+        ( lenSmall,
+          Override.modifyGlobal mvar $ \_mem ->
+            unimplemented "callGetHostName" Unimplemented.GetHostNameSmallSize,
+          Nothing
+        ),
+        -- TODO Check for name size
+        -- Otherwise, return a canned name
+        ( What4.truePred sym,
+          Override.modifyGlobal mvar $ \mem ->
+            liftIO $
+              do
+                let val = LLVMMem.LLVMValString hostname
+                let ty = LLVMMem.llvmValStorableType val
+                mem1 <- LLVMMem.storeRaw sym mem ptr ty noAlignment val
+                bv0' <- What4.bvLit sym (knownNat @32) (BV.mkBV (knownNat @32) 0)
+                return (bv0', mem1),
+          Nothing
+        )
+      ]
+
+-- | Super unsound. Returns the same string value every time, but in a fresh
+-- allocation for each call. A more sound approximation would be to require
+-- that the variable name be concrete and then have a map from names to
+-- allocations holding values.
```

Comment:
I think this is reasonable for now. In the future, it would be nice if we could just pull this functionality from DanM's work on symbolic IO https://github.com/GaloisInc/crucible/tree/feature/symio/crucible-symio.  That said, if we could feed information back into the values generated, that would become *very* interesting (and only suitable for uc-crux)

Reply:
> That said, if we could feed information back into the values generated

This is definitely a possibility, though as above I'm not exactly sure what we'd want to discover about the program and feed back in to this override.

Reply:
See more discussion above.